src/packages/admin-ui-components/src/table/component.tsx

```diff
@@ -206,15 +230,57 @@ export const Table = <T extends TableRowItem>({
 		[search, selectedEntity]
 	);
 
-	if (!selectedEntity) throw new Error('There should always be a selected entity at this point.');
-
 	if (loading) {
 		return <Loader />;
 	}
 	if (error) {
 		return <pre>{`Error! ${error.message}`}</pre>;
 	}
 
+	const handleSelectedRowsChange = (selectedRows: Set<string>) => {
+		setSelectedRows(selectedRows);
+	};
+
+	const handleDelete = () => {
+		setShowDeleteConfirmation(true);
+	};
+
+	const handleDeleteEntities = () => {
+		const ids = Array.from(selectedRows);
+
+		// For each id, call a separate delete mutation
+		const results = [];
+
+		for (const id of ids) {
+			// If it's the last one, then we can refetch
+			if (id === ids[ids.length - 1]) {
+				results.push(deleteEntity({ variables: { id }, refetchQueries: [`AdminUIListPage`] }));
```

Comment:
Could we use the `deleteMany` mutation here?

Reply:
The AC was specifically to call the delete mutation many times, and we talked about this in sprint planning. But yea I will try to implement the deleteMany, should be out of the box.

Reply:
Was not out of the box

Reply:
I think the other thing is even if it's the last deleteEntity call, they can still execute out of order, e.g.

Send to server requests to delete entity 1, then 2, then 3 (with refetch).

Requests 1 and 3 go to the same lambda, while 2 goes to a new one which has to do a cold boot, and therefore takes longer. The refetch on 3 includes 2 because it's not yet deleted.

To prevent the race condition in this scenario we'd need to wait for all 3 to finish, then call refetch ourselves once we've received confirmation that all requests are complete.